server/src/main/java/org/opensearch/common/cache/store/OpenSearchOnHeapCache.java

```diff
@@ -193,12 +211,18 @@ public String getCacheName() {
      */
     public static class Builder<K, V> extends ICacheBuilder<K, V> {
         private List<String> dimensionNames;
+        private boolean useNoopStats;
 
         public Builder<K, V> setDimensionNames(List<String> dimensionNames) {
             this.dimensionNames = dimensionNames;
             return this;
         }
 
+        public Builder<K, V> setUseNoopStats(boolean useNoopStats) {
+            this.useNoopStats = useNoopStats;
+            return this;
+        }
```

Comment:
Do we need this setter (and the helper method)?

Given that we intend to remove the feature flag in future, couldn't we just check the feature flag in the constructor?

We also don't need the `useNoopStats` field. The whole thing should just be "If the feature flag is enabled, use `DefaultCacheStatsHolder`, else use `NoopCacheStatsHolder`.

Reply:
Yes, if we always use cache.count() this works.